addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/protocol/RemoteControllerWebSocket.java

```diff
@@ -0,0 +1,678 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.samsungtv.internal.protocol;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.eclipse.jetty.util.component.LifeCycle.Listener;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.api.WebSocketAdapter;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.minidev.json.JSONArray;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+
+/**
+ * The {@link RemoteControllerWebSocket} is responsible for sending key codes to the
+ * Samsung TV via the websocket protocol (for newer TV's).
+ *
+ * @author Arjan Mels - Initial contribution
+ */
+public class RemoteControllerWebSocket extends RemoteController implements Listener {
+
+    private final Logger logger = LoggerFactory.getLogger(RemoteControllerWebSocket.class);
+
+    private final static String WS_ENDPOINT_REMOTE_CONTROL = "/api/v2/channels/samsung.remote.control";
+    private final static String WS_ENDPOINT_ART = "/api/v2/channels/com.samsung.art-app";
+    private final static String WS_ENDPOINT_V2 = "/api/v2";
+    private WebSocketClient client;
+
+    UUID uuid = UUID.randomUUID();

```

Comment:
Can those be private? Can they be `final`?

Reply:
Yes they can. Thanks for the catch.